apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryInsertRow.cs

```diff
@@ -197,54 +197,42 @@ internal TableDataInsertAllRequest.RowsData ToRowsData(bool allowEmptyInsertIds)
-                        // validate when we're given the list, as it can change.
-                        default:
-                            throw new InvalidOperationException("Local DateTime values are not supported");
-                    }
-                case DateTimeOffset dto:
-                    // Timestamp
-                    return dto.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture);
-                case TimeSpan ts:
-                    if (ts < TimeSpan.Zero || ts >= TimeSpan.FromHours(24))
-                    {
-                        throw new InvalidOperationException("TimeSpan values must be non-negative and less than 24 hours");
-                    }
-                    // This is ugly, but it avoids a trailing . when there are no fractional seconds.
-                    return (new DateTime(1970, 1, 1) + ts).ToString("HH:mm:ss.FFFFFF", CultureInfo.InvariantCulture);
-                case BigQueryNumeric numeric:
-                    return numeric.ToString();
-                case BigQueryBigNumeric bigNumeric:
-                    return bigNumeric.ToString();
-                case BigQueryGeography geography:
-                    return geography.ToString();
-                case BigQueryInsertRow row:
-                    return row.GetJsonValues();
-            }
+                // Civil datetime
+                DateTimeKind.Unspecified => dt.ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF", CultureInfo.InvariantCulture),
+                // Timestamp
+                DateTimeKind.Utc => dt.ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture),
+                // We can get into this situation if we accept a list of DateTime values. We can't
+                // validate when we're given the list, as it can change.
+                _ => throw new InvalidOperationException("Local DateTime values are not supported"),
+            },
+            DateTimeOffset dto => dto.ToUniversalTime().ToString("yyyy-MM-dd'T'HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture),
+            TimeSpan ts => ts switch
+            {
+                _ when ts < TimeSpan.Zero || ts >= TimeSpan.FromHours(24) => throw new InvalidOperationException("TimeSpan values must be non-negative and less than 24 hours"),
+                // This is ugly, but it avoids a trailing . when there are no fractional seconds.
+                _ => (new DateTime(1970, 1, 1) + ts).ToString("HH:mm:ss.FFFFFF", CultureInfo.InvariantCulture),
+            },
+            BigQueryNumeric numeric => numeric.ToString(),
+            BigQueryBigNumeric bigNumeric => bigNumeric.ToString(),
+            BigQueryGeography geography => geography.ToString(),
+            BigQueryInsertRow row => row.GetJsonValues(),
+            _ => GetValue(value),
+        };
+
+        private static object GetValue(object value)
```

Comment:
I think you can include these on the switch...?

Reply:
Hey Amanda, Post TimeSpan refactor to a method, please see if this is fine. Happy to change it if you think otherwise.